### PR TITLE
react-native-testing blog added

### PR DESCRIPTION
**Pull request by Saad-Bashar**

### CODE REVIEW

File: content/posts/2023-01-16-react-native-testing/index.md [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+    />,
+  );
+
+  const label = getByText("John Doe's Twitter");
+  fireEvent.press(label);
+  expect(Linking.openURL).toHaveBeenCalledWith('https://twitter.com/');
+});
+```
+
+Similarly to the previous test cases, let's break down the code. We are using the `fireEvent` method to fire the `press` event on the label. We are then using the `toHaveBeenCalledWith` matcher to check if the `Linking.openURL` method is called with the correct link. If the link is correct, the test will pass. If the link is not correct, the test will fail.
```
Comment (Savinvadim1312) [inline content/posts/2023-01-16-react-native-testing/index.md:356]: Maybe it's a good idea to remind people that the SocialLinks component, uses `Linking.openURL` when users click on it, that's why in our test, we are making sure that it was actually called. 

I think the confusion will be, "What does Linking has to do with this test", or "Do we always use Linking to test button clicks?"

File: content/posts/2023-01-16-react-native-testing/index.md [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+
+## GitHub repository
+
+You can find the GitHub repository for this article [here](https://github.com/notJust-dev/react-native-testing).
+
+## Conclusion
+
+We have now properly tested our SocialLinks component. We have made sure that the component is rendering the label and the icon of the social link. We have also made sure that the component is opening the correct link when the user clicks on the social link.
+
+Even though the test cases were fairly simple in this article, you can use the same approach to test more complex components as well. I hope you found this article will help you in your journey of learning React Native testing.
```
Comment (Savinvadim1312) [inline content/posts/2023-01-16-react-native-testing/index.md:392]: Reformulate the last sentence: "I hope you found this article will help you"

**Review by Savinvadim1312 — CHANGES_REQUESTED** [review]

File: content/posts/2023-01-16-react-native-testing/index.md [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+  View,
+  Text,
+  StyleSheet,
+  TouchableOpacity,
+  Linking,
+  Image,
+} from 'react-native';
+
+const Icons = {
+  twitter: require('../assets/twitter.png'),
```
[the remark this answers is not in the document]
Comment (Saad-Bashar) [reply content/posts/2023-01-16-react-native-testing/index.md:188]: Yeah they can get it from the github link. Adding a reference here.

File: content/posts/2023-01-16-react-native-testing/index.md [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+
+![react-native-testing-preview.png](./images/preview.png)
+
+## Writing the first test
+
+The component that we will be testing is the `<SocialLinks />`.It's always best to plan out the test. As a rule of the best practices, we should test the cases that our users see and interact. Let's dissect our `SocialLinks` component and see what are the cases that our users will see.
+
+1. The user will see the label of the social link.
+2. The user will see the icon of the social link.
+3. The user will see the social link is clickable.
```
[the remark this answers is not in the document]
Comment (Saad-Bashar) [reply content/posts/2023-01-16-react-native-testing/index.md:238]: Right. My bad! removing it.

File: content/posts/2023-01-16-react-native-testing/index.md (hunk 1 above)
Comment (Saad-Bashar) [reply content/posts/2023-01-16-react-native-testing/index.md:356]: Added why.

**Pull request merged**